test/unit/builtins/object.spec.ts

```diff
@@ -121,3 +121,93 @@ describe(".hasOwnProperty()", () => {
         `.expectToMatchJsResult();
     });
 });
+
+const trueFalseTests = [[true], [false]] as const;
+
+describe("Object.defineProperty", () => {
+    test.each(trueFalseTests)("writable (%p)", value => {
+        util.testFunction`
+            const foo = { bar: true };
+            Object.defineProperty(foo, "bar", { writable: ${value} });
+            let error = false;
+            try {
+                foo.bar = false;
+            } catch {
+                error = true;
+            }
+            return { foobar: foo.bar, error };
+        `.expectToMatchJsResult();
+    });
+
+    test.each(trueFalseTests)("configurable (%p)", value => {
+        util.testFunction`
+            const foo = { bar: true };
+            Object.defineProperty(foo, "bar", { configurable: ${value} });
+            try { delete foo.bar } catch {};
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test("defines a new property", () => {
+        util.testFunction`
+            const foo: any = {};
+            Object.defineProperty(foo, "bar", { value: true });
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test("overwrites an existing property", () => {
+        util.testFunction`
+            const foo = { bar: false };
+            Object.defineProperty(foo, "bar", { value: true });
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test("default descriptor", () => {
+        util.testFunction`
+            const foo = {};
+            Object.defineProperty(foo, "bar", {});
+            return Object.getOwnPropertyDescriptor(foo, "bar");
+        `.expectToMatchJsResult();
+    });
+
+    test.each([
+        ["{ value: true, get: () => true }"],
+        ["{ value: true, set: value => {} }"],
+        ["{ writable: true, get: () => true }"],
+    ])("invalid descriptor (%p)", props => {
+        util.testFunction`
+            const foo: any = {};
+            let err = false;
+
+            try {
+                Object.defineProperty(foo, "bar", ${props});
+            } catch {
+                err = true;
+            }
+
+            return { prop: foo.bar, err };
+        `.expectToMatchJsResult();
+    });
+});
+
+describe("Object.getOwnPropertyDescriptor", () => {
+    test("descriptor is exactly the same as the last one set", () => {
+        util.testFunction`
+            const foo = { bar: true };
+            Object.defineProperty(foo, "bar", {});
+            return Object.getOwnPropertyDescriptor(foo, "bar");
+        `.expectToMatchJsResult();
+    });
```

Comment:
Not sure if it belongs there or to class accessor test, but I think we need to test that **own**PropertyDescriptor doesn't include inherited descriptors.
```ts
class A { get a() { return true } }
Object.getOwnPropertyDescriptor(new A(), "a") // should be undefined
```

Reply:
Also maybe
```ts
class A { get a() { return true } }
Object.getOwnPropertyDescriptor(A.prototype, "a") // not undefined
```

Reply:
I can't quite replicate this behaviour, the metatable logic is getting the best of me. Would it be okay to bug this for a future fix?

Reply:
Sure, I was assuming it already worked like that